### PR TITLE
Add CLI interface for cortex command - Fixes #11

### LINKED ISSUE #11: Simple CLI Interface
**Title**: Basic command-line interface for Cortex

**Description**:
Create `cortex` command that users can run.

**Requirements**:
- Entry point: `cortex install <software>`
- Calls LLM integration layer
- Displays progress
- Shows results

**Example**:
```bash
$ cortex install docker
🧠 Understanding request...
📦 Planning installation...
⚙️ Installing Docker...
✅ Docker installed successfully
```

**Acceptance Criteria**:
- [ ] Works for basic install command
- [ ] User-friendly output
- [ ] Error messages helpful
- [ ] Installation via pip/setup.py

**Skills**: Python, CLI design, argparse

**Bounty**: $25 upon merge

### PR DESCRIPTION
**Pull request by Sahilbhatane**
# Simple CLI Interface

Fixes #11

## Summary
Implemented a command-line interface for Cortex that allows users to install software using natural language commands. The CLI integrates with the existing LLM layer to parse requests and generate validated system commands.

### Core Features
- **Entry Point**: `cortex install <software>` command
- **LLM Integration**: Calls CommandInterpreter to parse natural language requests
- **Progress Display**: Animated spinner and status indicators (🧠, 📦, ⚙️, ✅, ❌)
- **Multiple Modes**:
  - Default: Shows generated commands without executing
  - `--execute`: Runs the generated commands
  - `--dry-run`: Displays commands for review
- **Smart API Detection**: Automatically detects OPENAI_API_KEY or ANTHROPIC_API_KEY
- **Error Handling**: Clear, actionable error messages

### Files Created

**`cortex/cli.py`** - Main CLI implementation
- `CortexCLI` class with install command logic
- Progress indicators and user-friendly output
- Command execution with subprocess
- Comprehensive error handling

**`cortex/__init__.py`** - Package initialization
- Version information
- Main entry point export

**`cortex/test_cli.py`** - Test suite
- 22 unit tests covering all CLI functionality
- Mocked API calls and subprocess execution
- Edge cases and error scenarios

**`setup.py`** - Package configuration
- pip installable package
- Console script entry point: `cortex`
- Dependencies from LLM/requirements.txt

**`MANIFEST.in`** - Package manifest
- Includes all necessary files for distribution

**`.gitignore`** - Git ignore rules
- Excludes build artifacts and cache files

## Usage Examples

```bash
# Show help
cortex --help
cortex install --help

# Generate commands (default mode)
cortex install docker

# Execute commands
cortex install docker --execute

# Dry run mode
cortex install "nginx web server" --dry-run

# Complex installations
cortex install "python 3.11 with pip and virtualenv" --execute

# **Notes**- 
User need to have credit in there API key from provider, OPENAI give 5$ of free credit which can help in testing above command but only once.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

## Release Notes

* **New Features**
  * Interactive CLI for AI-assisted software installation with support for OpenAI and Claude providers
  * Dry-run mode to preview installation commands before execution
  * Command execution with progress tracking and verification
  * Updated default AI model selection for improved performance

* **Chores**
  * Added project packaging configuration and standard project ignores

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

**Comment by mikejmorgan-ai** [comment]
This is excellent - the CLI is the user's first interaction with Cortex Linux, so this is critical.

Before merge, need to validate:

1. **Does this integrate with the existing LLM layer (PR #5)?** The CommandInterpreter should be calling your merged code.

2. **API key handling** - How does it find ANTHROPIC_API_KEY or OPENAI_API_KEY? Environment variables?

3. **Testing** - Can you provide exact steps to test this locally?
pip install -r requirements.txt
export ANTHROPIC_API_KEY="..."
cortex install docker

4. **Error messages** - What happens if no API key is found? Does it give clear instructions?

Once I can verify this works end-to-end, this merges immediately.

**Re: Issue #11** - Was there an Issue #11 created, or should this be fixing a different issue number?

Also - you've now delivered:
- PR #5: LLM integration ($200 bounty) ✅
- PR #18: CLI interface (new work)

If this CLI PR is substantial, I'm adding another $150-200 bounty for it. Let me test it first.

Great work building the user experience layer.

**Comment by Sahilbhatane** [comment]
Yes ofcourse you can test it, I have considered all this and created a dedicated test file as well for it. For your simplicity and testing just run `test_cli.py`.

File: cortex/cli.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import sys
+import os
+import argparse
+import time
+from typing import List, Optional
+import subprocess
```
Comment (dhvll) [inline cortex/cli.py]: Remove unused import.

**Comment by dhvll** [comment]
Open issue #8 in separate PR. And make a habit of uploading video demonstration of your code is working. Just test cases are not enough. It becomes easier for reviewer to review the PR.

**Comment by mikejmorgan-ai** [comment]
Great progress! The core implementation is solid. To complete this PR, please:

1. **Integration** - Connect to main CLI (`cortex install [package]` flow)
2. **Documentation** - Complete README with usage examples
3. **End-to-End Test** - Add one test showing full user workflow
4. **Demo Video/GIF** - Show it working (huge bonus for visibility)

This is the most critical feature for Cortex Linux. Your work is laying the foundation for everything else. Keep going! 💪

Bounty remains $100 on merge. Additional $25 bonus if you include working demo.

**Comment by mikejmorgan-ai** [comment]
@dhvll 

Saw PR #18 (CLI interface) has a lot of discussion happening. This is your call as Lead

1. Review the code quality and implementation
2. Request changes if needed
3. Merge when it meets our standards
4. Reject if it doesn't fit our architecture

I trust your technical judgment. You're the decision-maker on all code reviews now.

Just keep me updated on:
- When major features merge (like this CLI)
- If there are any conflicts between contributors
- Bounty decisions (who gets paid for what)

Otherwise, you run the technical side. I'm focused on fundraising and business development.

- Mike
